### PR TITLE
Split out hal implementation to efm32hg-hal crate

### PR DESCRIPTION
**Pull request by fudanchii**
Try to leverage gpio and cmu hal implementation from efm32gg-hal.

- Remove some peripherals to be rewritten later
- Remove `enable_trace`, sadly dcb is not available in cortex-m0+
- Remove `MonoTimer`, sadly dwt lacks cyccnt support in cortex-m0+

~It compiles, but still not in runnable state.~ It compiles and all examples are running now.

on progress...

### CODE REVIEW

**Comment by jacobrosenthal** [comment]
Dont forget 
https://github.com/jacobrosenthal/efm32gg-hal/commit/adea5bd73fc085e7571ccb2d954cd7fae1cb3b42

**Comment by jacobrosenthal** [comment]
I also found that the bit banding doesnt seem to work, I temporarily tried outclr and outset implementation which worked
https://github.com/jacobrosenthal/efm32gg-hal/commit/42008ea48af7236dd8a5e66fd089927be217bb33

**Comment by jacobrosenthal** [comment]
This is where I got with those fixes
https://github.com/jacobrosenthal/imtomu-rs/tree/efm32-hal-delay

Since he doesnt wrap systick I had to have multiple delays to get to 1 second, which seems to work

**Comment by jacobrosenthal** [comment]
Also, moved watchdog over to a pr there https://github.com/chrysn/efm32gg-hal/pull/3 seemed to work

**Comment by fudanchii** [comment]
> I also found that the bit banding doesnt seem to work, I temporarily tried outclr and outset implementation which worked

Noted. bit-banding seems to not available at cortex M0+, confirmed at silabs' own knowledge base: 
https://www.silabs.com/community/mcu/32-bit/knowledge-base.entry.html/2016/04/04/understanding_periph-xqkx

I'll work on the gpio in a bit more, also would like to have all the [gpio modes](https://github.com/fudanchii/imtomu-rs/blob/master/src/gpio.rs#L30-L104) back for efm32hg.

I think @chrysn already stated some concerns related to this (the availability of gpio set/clear registers). We could discuss this further over there.

**Comment by fudanchii** [comment]
> We could discuss this further over there.
https://github.com/em32-rs/wg/issues/6

File: src/led.rs [hunk 1]
```diff
@@ -1,82 +1,41 @@
...
-    /// Mutably borrow red led.
-    /// see `green(&mut self)` documentation
-    /// for more info.
-    pub fn red(&mut self) -> &mut B7<OpenDrain> {
-        &mut self.red
-    }
-}
-
-/// Common trait for leds
-pub trait LedTrait {
```
Comment (jacobrosenthal) [inline src/led.rs]: Small nitpick. Can we get the LedTrait back. Tried to use it for something and noticed it was gone :)

Comment (fudanchii) [reply src/led.rs]: Added. sorry I took my time on this.

**Comment by fudanchii** [comment]
this closes #15

**Pull request merged**